app/src/main/java/chat/rocket/android/profile/presentation/ProfilePresenter.kt

```diff
@@ -1,72 +1,141 @@
 package chat.rocket.android.profile.presentation
 
+import android.graphics.Bitmap
+import android.net.Uri
+import chat.rocket.android.chatroom.domain.UriInteractor
 import chat.rocket.android.core.behaviours.showMessage
 import chat.rocket.android.core.lifecycle.CancelStrategy
+import chat.rocket.android.helper.UserHelper
 import chat.rocket.android.server.domain.GetCurrentServerInteractor
 import chat.rocket.android.server.infraestructure.RocketChatClientFactory
+import chat.rocket.android.util.extension.compressImageAndGetByteArray
 import chat.rocket.android.util.extension.launchUI
 import chat.rocket.android.util.extensions.avatarUrl
 import chat.rocket.android.util.retryIO
 import chat.rocket.common.RocketChatException
 import chat.rocket.common.util.ifNull
 import chat.rocket.core.RocketChatClient
-import chat.rocket.core.internal.rest.me
+import chat.rocket.core.internal.rest.resetAvatar
 import chat.rocket.core.internal.rest.setAvatar
 import chat.rocket.core.internal.rest.updateProfile
+import java.util.*
 import javax.inject.Inject
 
 class ProfilePresenter @Inject constructor(
     private val view: ProfileView,
     private val strategy: CancelStrategy,
+    private val uriInteractor: UriInteractor,
+    val userHelper: UserHelper,
     serverInteractor: GetCurrentServerInteractor,
     factory: RocketChatClientFactory
 ) {
     private val serverUrl = serverInteractor.get()!!
     private val client: RocketChatClient = factory.create(serverUrl)
-    private lateinit var myselfId: String
+    private val myselfId = userHelper.user()?.id ?: ""
```

Comment:
Don't need to change now, but you can use `.orEmpty()` like:
`private val myselfId = userHelper.user()?.id.orEmpty()`

Reply:
Got it! 
Helpful! Thank you!